### DISCUSSION TITLE
Discuss Uniswap

### DISCUSSION THREAD

**Discussion by utoexo** [knightsven/near-smart-contract-rust-template#4] [Q&A] [answered]
What is the difference between V3 and V4?

**Comment by magicbroomstick** [comment, marked as answer]
While Uniswap V3 gives each pool its own contract, in Uniswap V4 all DeFi pools go into a single contract. This should make the creation of a new pool 99% cheaper. Crucial is the role of the Pool Manager, a piece of software that maintains and maps all different pools.

**Comment by SteveK0673** [comment]
gfbdfg fdgn
